airbyte-integrations/connectors/source-mysql/src/test/java/io/airbyte/integrations/source/mysql/CdcMySqlSourceTest.java

```diff
@@ -257,6 +259,88 @@ public void tearDown() {
     }
   }
 
+  @Test
+  public void fullRefreshAndCDCShouldReturnSameRecords() throws Exception {
+    setupForComparisonBetweenFullRefreshAndCDCSnapshot();
+
+    AirbyteCatalog discover = source.discover(config);
+    List<AirbyteStream> streams = discover.getStreams();
+
+    assertEquals(streams.size(), 1);
+    JsonNode jsonSchema = streams.get(0).getJsonSchema().get("properties");
+    assertEquals(jsonSchema.get("id").get("type").asText(), "number");
+    assertEquals(jsonSchema.get("bool_col").get("type").asText(), "boolean");
+    assertEquals(jsonSchema.get("tiny_int_one_col").get("type").asText(), "boolean");
+    assertEquals(jsonSchema.get("tiny_int_two_col").get("type").asText(), "number");
+
+    AirbyteCatalog catalog = new AirbyteCatalog().withStreams(streams);
+    final ConfiguredAirbyteCatalog configuredCatalog = CatalogHelpers
+        .toDefaultConfiguredCatalog(catalog);
+    configuredCatalog.getStreams().forEach(c -> c.setSyncMode(SyncMode.FULL_REFRESH));
+
+    Set<JsonNode> dataFromFullRefresh = extractRecordMessages(
+        AutoCloseableIterators.toListAndClose(source.read(config, configuredCatalog, null)))
+            .stream()
+            .map(AirbyteRecordMessage::getData).collect(Collectors.toSet());
+
+    configuredCatalog.getStreams().forEach(c -> c.setSyncMode(SyncMode.INCREMENTAL));
+    Set<JsonNode> dataFromDebeziumSnapshot = extractRecordMessages(
+        AutoCloseableIterators.toListAndClose(source.read(config, configuredCatalog, null)))
+            .stream()
+            .map(
+                airbyteRecordMessage -> {
+                  JsonNode data = airbyteRecordMessage.getData();
+                  removeCDCColumns((ObjectNode) data);
+                  /**
+                   * Debezium reads TINYINT (expect for TINYINT(1)) as IntNode while FullRefresh reads it as Short Ref
+                   * : {@link io.airbyte.db.jdbc.JdbcUtils#setJsonField(java.sql.ResultSet, int, ObjectNode)} -> case
+                   * TINYINT, SMALLINT -> o.put(columnName, r.getShort(i));
+                   */
+                  ((ObjectNode) data)
+                      .put("tiny_int_two_col", (short) data.get("tiny_int_two_col").asInt());
+                  return data;
+                })
+            .collect(Collectors.toSet());
+
+    JsonNode data1 = Jsons.jsonNode(ImmutableMap.of(
```

Comment:
just a small comment for maintainability: can we return these values from (or input them to) the `setupforComparison` method so they are not defined in separate places like this?